ci: github: Update for deprecation of add-path (backport some fixes to doc/zephyr.doxyfile.in)

galak opened the pull request:
Github has deprecated add-path, so update the workflows that use it to
the new method of setting the PATH.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

nashif committed:
docs: remove deprecated doxygen options

Noticed those with most recent doxygen, we do not use those options, so
it is safe to remove now.

[Backport of f2b04d9ac2865be10bfe694ba1dfce7f3c23f68c]

Signed-off-by: Anas Nashif <anas.nashif@intel.com>
Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak committed:
doc: Fix warnings associated with 'unbalanced grouping commands'

Builds of docs with doxygen 1.8.16 has a number of warnings of the form:
'warning: unbalanced grouping commands'.  Fix those warnings be either
balancing the group command or removing it.

[Backport of a8171db6a6c8a06e87446e25b55357ad1ba97719]

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

mbolivar-nordic committed:
doc: fix some errors

I've been seeing these cause errors on the more recent versions of
Doxygen which come with Arch Linux for a while now. Fix these:

error: Illegal format for option TCL_SUBST, no equal sign ('=') specified for item 'YES'
$ZEPHYR_BASE/tests/kernel/mem_protect/futex/src/main.c:461: warning: end of file with unbalanced grouping commands

Just trying to get them out of my local output and as preparation for
whenever they start showing up for Ubuntu.

[Backport of ededeb2736d0b4da2d65d79966ccfc51b16a70b1]

Signed-off-by: Martí Bolívar <marti.bolivar@nordicsemi.no>

galak committed:
ci: github: Update for deprecation of add-path

Github has deprecated add-path, so update the workflows that use it to
the new method of setting the PATH.

[Backport of fabf40398fe92e2ebffe03fd4628d82d43e2c1b5]

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

joerchan commented:
Not sure I'm the best to approve this, I don't know this part.

Is this a backport of an existing commit in master? Maybe add a reference to it?

galak commented:
> Not sure I'm the best to approve this, I don't know this part.
> 
> Is this a backport of an existing commit in master? Maybe add a reference to it?

Add references in the commit messages.

galak commented:
@gmarull any thoughts on what's going on with the docs build issue with tests/kernel/common/src/intmath.c

gmarull commented:
@galak I can't see anything suspicious on that file. Does it only happen on *new* Doxygen versions?

galak commented:
> @galak I can't see anything suspicious on that file. Does it only happen on _new_ Doxygen versions?

No idea, whatever version of doxygen is getting used by the ci check.  Looks like 1.8.17

nashif commented:
maybe we should just exclude doxygen building tests in the branch now and get this going:

```
diff --git a/doc/zephyr.doxyfile.in b/doc/zephyr.doxyfile.in
index 49d78b92cf..e63a8ae069 100644
--- a/doc/zephyr.doxyfile.in
+++ b/doc/zephyr.doxyfile.in
@@ -762,8 +762,7 @@ INPUT                  = @ZEPHYR_BASE@/include/ \
                          @ZEPHYR_BASE@/include/arch/nios2/ \
                          @ZEPHYR_BASE@/lib/libc/minimal/include/ \
                          @ZEPHYR_BASE@/include/net/dns_resolve.h \
-                        @ZEPHYR_BASE@/subsys/testsuite/ztest/include/ \
-                        @ZEPHYR_BASE@/tests/kernel/
+                        @ZEPHYR_BASE@/subsys/testsuite/ztest/include/

```

galak commented:
> maybe we should just exclude doxygen building tests in the branch now and get this going:

No issue w/me, what implication does this have when we release the next v1.14.x version for docs on release?

nashif commented:
> No issue w/me, what implication does this have when we release the next v1.14.x version for docs on release?

none, we do not link the tests in documentation, it is only available in doxygen output and was added just to make sure structure is correct. We can also fix that later, right now however this will unblock all those 1.14 PRs

galak committed:
doc: Drop tests/kernel from doxygen input path

We are running into a doxygen error with intmath.c from
tests/kernel/common/src/.

intmath.c:29: warning: member with no name found.

Drop tests/kernel from the doxygen input path for now.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak commented:
> none, we do not link the tests in documentation, it is only available in doxygen output and was added just to make sure structure is correct. We can also fix that later, right now however this will unblock all those 1.14 PRs

Pushed that change into the PR.
